shared_model/interfaces/common_objects/amount.hpp

```diff
@@ -59,10 +55,9 @@ namespace shared_model {
       Amount *clone() const override;
 
      private:
-      const std::string amount_;
-
+      std::string string_repr_;
```

Comment:
it makes me nervous a bit that we actually have 2 different backend for a single object. It makes possible (in theory) desynchronisation between. Maybe just recalculate `string_repr_ ` on fly? I could have a negative performance impact but I don't think it's big.

Reply:
I see. What if I make both representations const?

Reply:
It would be definitely better.